parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -910,4 +913,28 @@ oras_login_with_kubelet_identity() {
     echo "successfully logged in to acr '$acr_url' with identity token"
 }
 
+ensureSSHService() {
+    # If not Ubuntu, no changes needed
+    if [ "$OS" = "$UBUNTU_OS_NAME" ]; then
+        # for Ubuntu 22.10+ or newer uses socket activation
+        if semverCompare "$OS_VERSION" "22.10"; then
+            # For Ubuntu 22.10+ with socket-based SSH activation
+            if systemctl is-active --quiet ssh.socket; then
+                # Socket is active, disable it and switch to service-based activation
+                systemctl disable --now ssh.socket
+                # Remove the socket-based configuration file if present
+                if [ -f /etc/systemd/system/ssh.service.d/00-socket.conf ]; then
+                    rm /etc/systemd/system/ssh.service.d/00-socket.conf
```

Comment:
this is a lot of indentation, would prefer an "exit-fast" approach  instead, something like:

```
if [ "$OS" != "$UBUNTU_OS_NAME" ]; then
  return 0
fi
if ! semverCompare "$OS_VERSION" "22.10"; then
  return 0
fi
if ! systemctl is-active --quiet ssh.socket; then
  return 0
fi

systemctl disable --now ssh.socket

....

systemctlEnableAndStart ssh 30 || return $ERR_SYSTEMCTL_START_FAIL

```